### ISSUE TITLE
DDoc extraction runs too often

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#4497]
ddoc extraction is supposed to update the secondary ddocs when the primary ddoc is updated and the compiled.json attachment has updated code. It seems to be running too often. This was captured during a build when running integration tests:

```
  migrations integration test framework
Updating docs: _design/medic-admin
    ✓ should be able to match docs without a fixed ID (2482ms)
Updating docs: _design/medic-admin
    ✓ should be able to match docs with or without a fixed ID (2441ms)
Updating docs: _design/medic-admin
    ✓ should handle a simple migration (2442ms)
    no-op
Updating docs: _design/medic-admin
      ✓ should leave an empty db empty (2461ms)
Updating docs: _design/medic-admin
      ✓ should leave a full db full (2751ms)
```

**Comment by garethbowen** [comment]
Code review please @dianabarsan

**Comment by ngaruko** [comment]
Does this need any AT-ing @garethbowen ?

**Comment by garethbowen** [comment]
Yes. Make sure ddoc extraction only runs when ddocs have actually changed.

**Comment by ngaruko** [comment]
LGTM -  ddoc extraction only runs when ddocs have actually changed. Only this one runs every tie we deploy : ` Updating docs: appcache`. I assume that's expected @garethbowen ?

**Comment by garethbowen** [comment]
@ngaruko The line should only be emitted when the `manifest.appcache` has changed. If you deploy a ddoc with the same `manifest.appcache` attachment over and over again it shouldn't be logged. Can you confirm you were only seeing that line when the attachment changed and not all the time?

**Comment by ngaruko** [comment]
Looks like `manifest.appcache` is (re-)created on deploy @garethbowen ?
```Running "appcache:inbox" (appcache) task
|  AppCache manifest "manifest.appcache" created.```

**Comment by garethbowen** [comment]
After thinking about this some more I think this is a byproduct of how we build the appcache manifest. It gets changed every time we build whether or not any files have actually changed. The way to fix this is to replace appcache with service workers which will happen sometime hopefully soon! I've raised #4713 to deal with this.

I think this should pass AT because the original issue has been resolved.
